kf_lib_data_ingest/etl/extract/extract.py

```diff
@@ -37,17 +37,38 @@ def lcm(number_list):
     )
 
 
+def clean_walk(dir):
+    paths = []
+    for root, directories, filenames in os.walk(dir):
+        for filename in filenames:
+            paths.append(os.path.join(root, filename))
+        toremove = []
+        for dir in directories:
+            if dir.startswith('.'):
+                toremove.append(dir)
+        for r in toremove:
+            directories.remove(r)
+    return paths
+
+
 class ExtractStage(IngestStage):
     def __init__(
-        self, stage_cache_dir, extract_config_paths, auth_configs=None
+        self, stage_cache_dir, extract_config_dir, auth_configs=None
     ):
         super().__init__(stage_cache_dir)
-        if isinstance(extract_config_paths, list):
-            self.extract_configs = [ExtractConfig(config_filepath)
-                                    for config_filepath
-                                    in extract_config_paths]
-        elif isinstance(extract_config_paths, str):
-            self.extract_configs = [ExtractConfig(extract_config_paths)]
+
+        assert_safe_type(extract_config_dir, str)
+        self.extract_config_dir = extract_config_dir
+        self.extract_configs = [
```

Comment:
Can't you just do this? It seems to work
```python
        self.extract_configs = []
        for root, directories, filenames in os.walk(extract_config_dir):
            for filename in filenames:
                if os.path.splitext(filename)[-1] == '.py':
                    ec = ExtractConfig(
                        os.path.join(root, filename)
                    )
                    ec.config_file_relpath = os.path.relpath(
                        ec.config_filepath,
                        start=self.extract_config_dir
                    )
                    self.extract_configs.append(ec)
```
Then you don't have to iterate multiple times either

Reply:
this won't avoid hidden nasties